### PR TITLE
Smartcontracts integration

### PR DESCRIPTION
**Pull request by krzkaczor**
- [x] initialize smart contract structure
- [x] add ethereum address to the wallet metadata
- [x] get all wallet data

### CODE REVIEW

File: app/modules/contracts/ContractsService.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { inject, injectable } from "inversify";
```
Comment (huesforalice) [inline app/modules/contracts/ContractsService.ts]: ContractsService.ts should be in the lib folder

File: app/components/dashboard/myWallet/MyWalletWidget.tsx [hunk 2]
```diff
@@ -41,50 +118,32 @@ export const MyWalletWidget: React.SFC<IMyWalletWidget> = ({
...
   );
 };
+
+export const MyWalletWidget = compose<React.SFC>(
+  onEnterAction({ actionCreator: d => d(actions.wallet.startLoadingWalletData()) }),
+  appConnect<StateProps>({
+    stateToProps: s => {
+      const isLoading = s.wallet.loading;
+      if (!isLoading) {
+        const walletData = s.wallet as IWalletStateLoaded; // TS doesnt realize here that we are sure that its wallet already loaded
```
Comment (huesforalice) [inline app/components/dashboard/myWallet/MyWalletWidget.tsx]: Won't this fail in case there is an error loading?

File: app/modules/init/sagas.ts [hunk 3]
```diff
@@ -4,22 +4,27 @@ import { TGlobalDependencies } from "../../di/setupBindings";
...
   try {
+    yield neuCall(detectUserAgent);
     yield web3Manager.initialize();
 
     const jwt = yield neuCall(loadJwt);
     if (jwt) {
       yield loadUser();
     }
     yield neuCall(loadPreviousWallet);
+    yield neuCall(initializeContracts);
```
Comment (huesforalice) [inline app/modules/init/sagas.ts]: This seems to me built in a way, that when trying to initialize the contracts fails, the whole app won't start. There will be quite a few requests to ethereum to set up all the contracts no? Maybe this should be a bit more robust.

Comment (krzkaczor) [reply app/modules/init/sagas.ts]: I think this is what we are going to do — if app won't connect to blockchain it's not usable. It seems that there will be a lot of views with data coming both from API and blockchain.

File: app/components/dashboard/myWallet/MyWalletWidget.tsx (hunk 2 above)
Comment (krzkaczor) [reply app/components/dashboard/myWallet/MyWalletWidget.tsx]: yeah... error handling is missing. Let me fix that

**Comment by krzkaczor** [comment]
@huesforalice i fixed your comments. And now it should be clear how to run it ;) please let me know what do you think

**Comment by huesforalice** [comment]
please add NF_UNIVERSE_CONTRACT_ADDRESS to .env file

**Pull request merged**